### PR TITLE
Documaster API: draft 16

### PR DESCRIPTION
**Pull request by ivaylomitrev**
Changes in the API documentation:

* Add information about Advanced free-text search

### CODE REVIEW

**Review by venelin-kotsev — APPROVED** [review]

File: v1/endpoints.md [hunk 1]
```diff
@@ -881,6 +882,21 @@ We recommend taking the following precautions when performing free-text searches
...
 
+### Advanced free-text search
+
+By specifying the `enableAdvancedFTS` flag, certain characters will gain special meaning.
+
+- Double quote (") allow you to perform phrase searches for exact matches. For example: "This is my phrase search that requires an exact match"
+- Asterisk (*) and question mark (?) allow you to perform wildcard searches.
+  - Asterisks match zero or more consecutive characters at the beginning, middle, or end of a string. For example: \*master, Documas\*, Doc\*ter
+  - Question marks match a single character at the beginning, middle, or end of a string. For example: ?ocumaster, Documaste?, Do?umaster
+- Tilde (~) allow you to perform fuzzy search and proximity search.
```
Comment (vulkodava) [inline v1/endpoints.md]: Here the word _allow_ should be _allows_.

Comment (ivaylomitrev) [reply v1/endpoints.md]: Fixed. I've also fixed "Double quoteS" above.

**Review by vulkodava — APPROVED** [review]

**Comment by phillipj** [comment]
Any chance this is somewhat related to solr's capabilities? If so, maybe worth referencing some of their docs?

**Comment by ivaylomitrev** [comment]
@phillipj , your comment wasn't on any of the files and I couldn't reply to it directly... GitHub oddity 
I guess.

Anyway, to respond to you - I did not link to Solr on purpose, because I did not want to expose information about the underlying implementation, thus, avoiding confusion. We are dependent on a very particular version of that software now and this documentation may no longer be true when we upgrade or even switch to a different underlying implementation. We'd be dependent on an external link then as well.

I just didn't consider it important enough to introduce the need of maintaining information about Solr in a (mostly) unrelated API, because it's easy to forget about it.

There are drawbacks to this though - we have to be more explicit in our own documentation and examples.

**Comment by phillipj** [comment]
Alrighty, thanks for sharing your rationale!

I've struggled with similar scenarios before, where trying to communicate the features with all its aspects and edge cases in a really good way becomes a challenge. Not to mention maintaining and improving it over time. Seen that neglected more than a few times to say the least, so I've tried finding other ways to deal with that and offloading to good official 3rd party docs (if they exist) has been a viable option at times.

**Pull request merged**